src/main/java/com/marklogic/client/ext/schemasloader/impl/QbvDocumentFileProcessor.java

```diff
@@ -0,0 +1,143 @@
+import java.util.List;
+
+public class QbvDocumentFileProcessor extends LoggingObject implements DocumentFileProcessor {
+
+	public static final String QBV_COLLECTION = "http://marklogic.com/xdmp/qbv";
+	public static final String QBV_XML_PLAN_NAMESPACE = "http://marklogic.com/plan";
+	public static final String QBV_XML_ROOT_ELEMENT = "query-based-view";
+	public static final String JAVASCRIPT_EVAL_TEMPLATE = "declareUpdate(); xdmp.invokeFunction(function() {'use strict'; const op = require('/MarkLogic/optic'); return %s }, {database: xdmp.database('%s')})";
+	public static final String XQUERY_EVAL_TEMPLATE = "xquery version \"1.0-ml\"; import module namespace op=\"http://marklogic.com/optic\" at \"/MarkLogic/optic.xqy\"; xdmp:invoke-function(function() {%s},<options xmlns=\"xdmp:eval\"><database>{xdmp:database('%s')}</database></options>)";
+
+	final private DatabaseClient databaseClient;
+	final private String qbvGeneratorDatabaseName;
+	final protected List<DocumentFile> qbvFiles = new ArrayList<>();
+	final private XMLDocumentManager docMgr;
+
+
+	/**
+	 * @param databaseClient - a MarkLogic DatabaseClient object for the Schemas database
+	 */
+	public QbvDocumentFileProcessor(DatabaseClient databaseClient, String qbvGeneratorDatabaseName) {
+		this.databaseClient = databaseClient;
+		this.qbvGeneratorDatabaseName = qbvGeneratorDatabaseName;
+		this.docMgr = databaseClient.newXMLDocumentManager();
+	}
+
+	@Override
+	public DocumentFile processDocumentFile(DocumentFile documentFile) {
+		if (isQbvQuery(documentFile)) {
+			// Defer processing until after all other schemas have been processed.
+			qbvFiles.add(documentFile);
+			return null;
+		} else {
+			return documentFile;
+		}
+	}
+
+	private boolean isQbvQuery(DocumentFile documentFile) {
+		String uri = documentFile.getUri();
+		String extension = documentFile.getFileExtension();
+		if (extension != null) {
+			extension = extension.toLowerCase();
+		}
+		return
+			"sjs".equals(extension)
+				|| "js".equals(extension)
+				|| "xqy".equals(extension)
+				|| "xq".equals(extension)
+				|| (uri != null && uri.startsWith("/qbv"));
+	}
+
+	public void processQbvFiles() {
+		qbvFiles.forEach(this::processQbvFile);
+		qbvFiles.clear();
+	}
+
+	public void processQbvFile(DocumentFile qbvFile) {
+		ServerEvaluationCall call = getServerEvaluationCall(qbvFile);
+		if (call != null) {
+			StringHandle handleString = new StringHandle();
+			try {
+				call.eval(handleString);
+				if (Format.XML.equals(handleString.getFormat())) {
+					Element root = new SAXBuilder().build(new StringReader(handleString.get())).getRootElement();
+					if (QBV_XML_ROOT_ELEMENT.equals(root.getName()) & (root.getNamespace() != null && root.getNamespace().getURI().equals(QBV_XML_PLAN_NAMESPACE))) {
+						qbvFile.getDocumentMetadata().getCollections().add(QBV_COLLECTION);
+						String uri = qbvFile.getUri() + ".xml";
+						DOMHandle handleXML = new DOMHandle(new DOMOutputter().output(root.getDocument()));
```

Comment:
The Java Client has a JDOMHandle that you can use so you don't need to do anything with DOM. You just need to get a reference to the Document that's returned by SAXBuilder.build above. So something like this:

```
Document doc = new SAXBuilder()....
Element root = doc.getRootElement();
```

And then a little further down:

```
docMgr.write(uri, qbvFile.getDocumentMetadata(), new JDOMHandle(xmlDocument));
```